### PR TITLE
Fix YouTube Regrets UI bug and script bug

### LINKED ISSUE #4615: Youtube Regrets: Join Mozilla Button
**Describe the bug**
The Join Mozilla button appears throughout the page and is not clickable. On smaller viewports, two buttons appear at the top, both not clickable.

**To Reproduce**
1. Go to 'https://foundation.mozilla.org/en/campaigns/youtube-regrets/'
2. Scroll down or minimize window (or mobile)

**Expected behavior**
See button after opening message only and is actionable to join Mozilla.

**Screenshots**
Y0yn5VGrgT

**Desktop (please complete the following information):**
 - OS: iOS 10.15.4
 - Browser: Firefox 75 & Chrome 81.0.4044.138

**Smartphone (please complete the following information):**
 - Device: Pixel 2
 - OS: android 10
 - Browse: Firefox 68.8.0 & Chrome 81.0.4044.138

### PR DESCRIPTION
**Pull request by mmmavis**
Closes #4615

Turned out the sign up button on YouTube Regrets page has been broken since my JS refactor work 😱 I'll file a ticket to add interaction testing for sign up forms.

### CODE REVIEW

File: network-api/networkapi/wagtailpages/templates/wagtailpages/pages/youtube_regrets_page.html [hunk 1]
```diff
@@ -33,8 +33,8 @@
       {% endfor %}
     </div>
     <div class="intro-button-wrapper">
-      <button class="btn btn-primary btn-newsletter d-lg-none mx-auto d-none">{% trans "Join Mozilla" %}</button>
-      <button class="btn btn-primary btn-newsletter d-none d-lg-block mx-auto d-none">{% trans "Join Mozilla" %}</button>
```
Comment (mmmavis) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/pages/youtube_regrets_page.html:37]: We conditionally toggle `d-none` in the script depending on how far the page is scrolled. The new Bootstrap class combo `d-none d-lg-block` are creating issues with the script we set up. For simplicity, I'm replacing them with custom SCSS code.

File: network-api/networkapi/wagtailpages/templates/wagtailpages/pages/youtube_regrets_page.html [hunk 2]
```diff
@@ -87,7 +87,3 @@
 {% block footer_block %}
 {% include "partials/footer.html" %}
 {% endblock %}
-
-{% block extra_scripts %}
-  <script src="{% static "_js/youtube-regrets.compiled.js" %}"></script>
```
Comment (mmmavis) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/pages/youtube_regrets_page.html:92]: This was a bad move. This script needs to wait for various functions and calls in `main.js` to finish first.

File: network-api/networkapi/wagtailpages/templates/wagtailpages/pages/youtube_regrets_page.html (hunk 1 above)
Comment (Pomax) [reply network-api/networkapi/wagtailpages/templates/wagtailpages/pages/youtube_regrets_page.html:37]: oh man well spotted

File: network-api/networkapi/wagtailpages/templates/wagtailpages/pages/youtube_regrets_page.html (hunk 2 above)
Comment (Pomax) [reply network-api/networkapi/wagtailpages/templates/wagtailpages/pages/youtube_regrets_page.html:92]: Hm, does it? I thought it was only dependent on the DOM being done, in which case we're only missing the `defer` attribute.

File: source/js/main.js [hunk 3]
```diff
@@ -64,6 +65,7 @@ let main = {
       this.injectReactComponents();
       this.bindHandlers();
       initializePrimaryNav(networkSiteURL, csrfToken, primaryNav);
+      this.initPageSpecificScript();
```
Comment (Pomax) [inline source/js/main.js:68]: As a generic function, we probably want to move this call to line 72, so that it runs after React is done.

Comment (mmmavis) [reply source/js/main.js]: Updated!

**Pull request merged**